src/util.js

```diff
@@ -28,7 +30,7 @@ const decoder = new cbor.Decoder({
     [CID_CBOR_TAG]: (val) => {
       // remove that 0
       val = val.slice(1)
-      return {'/': val}
+      return new CID(val)
```

Comment:
Until https://github.com/ipld/ipld/issues/44 is solved, it should return `{ '/': new CID(val) }`, fixing only one of the issues which that the CIDs were coming as buffers, making it painful for users to cast it themselves.

Reply:
Then I would wait with a merge until https://github.com/ipld/ipld/issues/44 is resolved. `{'/': CID()}` sounds to me like the worst of both worlds. It's hard to explain. I would then rather change the return value from Buffer to a string.